### PR TITLE
feat (ui): workflow trigger (add/update) + display context + delete node

### PR DESCRIPTION
**Pull request by sguiheux**

### CODE REVIEW

File: ui/src/app/service/workflow/workflow.store.ts [hunk 1]
```diff
@@ -84,4 +84,36 @@ export class WorkflowStore {
...
+        return this._workflowService.deleteWorkflow(key, workflow).map(w => {
+            let workflowKey = key + '-' + workflow.name;
+            let store = this._workflows.getValue();
+            this._workflows.next(store.delete(workflowKey));
+            return w;
+        });
+    }
+
+    getTriggerCondition(key: string, workflowName: string, nodeID: number): Observable<WorkflowTriggerConditionCache> {
+        return this._workflowService.getTriggerCondition(key, workflowName, nodeID).map(wtc => wtc);
```
Comment (bnjjj) [inline ui/src/app/service/workflow/workflow.store.ts]: I think `.map(wtc => wtc)`is useless in this case

File: ui/src/app/shared/workflow/node/workflow.node.html [hunk 2]
```diff
@@ -1,3 +1,23 @@
 <div class="workflowNode">
-    {{node.pipeline.name}}
-</div>
\ No newline at end of file
+    <div class="ui segment">
+        <div class="pipeline">
+            <p class="ellipsis">{{node.pipeline.name + '(' + node.id + ')'}}fdsfd d</p>
```
Comment (bnjjj) [inline ui/src/app/shared/workflow/node/workflow.node.html]: `fdsfd d` typo ?

File: ui/src/app/shared/workflow/trigger/condition-list/trigger.condition.list.component.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    getData(): any[] {
+        return this.conditions;
+    }
+
+    removeCondition(cond: WorkflowTriggerCondition): void {
+        let indexToDelete = this.conditions.findIndex(c => c.variable === cond.variable);
+        if (indexToDelete !== -1) {
+            this.conditions.splice(indexToDelete, 1);
```
Comment (bnjjj) [inline ui/src/app/shared/workflow/trigger/condition-list/trigger.condition.list.component.ts]: I think in this case a `.filter(...)` should be better for complexity

File: ui/src/app/views/application/show/workflow/application.workflow.html [hunk 4]
```diff
@@ -47,7 +39,9 @@
          [class.horizontal]="workflowOrientation === 'horizontal'"
```
Comment (bnjjj) [inline ui/src/app/views/application/show/workflow/application.workflow.html]: `[ngClass]="workflowOrientation"`  ?

File: ui/src/app/views/workflow/show/workflow.component.ts [hunk 5]
```diff
@@ -125,36 +168,88 @@ export class WorkflowShowComponent implements AfterViewInit {
...
+    private openEditTriggerModal(parentID: string, childID: string) {
+        let pID = Number(parentID.replace('node-', ''));
+        let cID = Number(childID.replace('node-', ''));
+        let node = Workflow.getNodeByID(pID, this.detailedWorkflow);
+        if (node && node.triggers) {
+            node.triggers.forEach(t => {
+                if (t.workflow_dest_node_id === cID) {
+                    this.selectedNode = _.cloneDeep(node);
+                    this.selectedTrigger = _.cloneDeep(t);
+                    return;
```
Comment (bnjjj) [inline ui/src/app/views/workflow/show/workflow.component.ts]: Continue to iterate

Comment (bnjjj) [inline ui/src/app/views/workflow/show/workflow.component.ts]: Use .some instead for this case maybe https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some?v=control

File: ui/src/app/views/application/show/workflow/application.workflow.html (hunk 4 above)
Comment (sguiheux) [inline ui/src/app/views/application/show/workflow/application.workflow.html]: maybe but this code will be deleted after workflow :)

**Pull request merged**